cosmo_tester/framework/cfy_helper.py

```diff
@@ -97,6 +101,34 @@ def bootstrap(self,
                 task_retry_interval=task_retry_interval,
                 verbose=verbose,
                 debug=debug).wait()
+            self.upload_plugins(cfy)
```

Comment:
I can't see where it does belong then, since it is needed on every bootstrap, and every bootstrap uses this. Moving it to the testenv bootstrap method will mean that we have to arbitrarily call the upload_plugins method in some tests, but not others.